conans/client/manager.py

```diff
@@ -110,21 +110,41 @@ def __init__(self, client_cache, user_io, runner, remote_manager, search_manager
         self._search_manager = search_manager
         self._settings_preprocessor = settings_preprocessor
 
-    def load_consumer_conanfile(self, conanfile_path, info_folder, output, reference=None,
-                                deps_info_required=False):
-
+    def _load_consumer_conanfile(self, conanfile_path, info_folder, output,
+                                 deps_info_required=False):
+        """loads a conanfile for local flow: source, imports, package, build
+        """
         profile = read_conaninfo_profile(info_folder) or self._client_cache.default_profile
         loader = self.get_loader(profile)
         if conanfile_path.endswith(".py"):
-            consumer = not reference
-            conanfile = loader.load_conan(conanfile_path, output, consumer, reference)
+            conanfile = loader.load_conan(conanfile_path, output, consumer=True)
         else:
             conanfile = loader.load_conan_txt(conanfile_path, output)
         if deps_info_required is not None:
             _load_deps_info(info_folder, conanfile, required=deps_info_required)
 
         return conanfile
 
+    def _load_install_conanfile(self, loader, reference_or_path, conanfile_filename, cwd=None):
+        """loads a conanfile for installation: install, info
+        cwd only used for virtuals, to pass it the current directory and make available the
+        conanfile.conanfile_directory (smell)"""
+        if isinstance(reference_or_path, ConanFileReference):
+            conanfile = loader.load_virtual([reference_or_path], cwd)
+            loader.dev_reference = reference_or_path
```

Comment:
So `conan install ref` should consider `ref` as a package being developed?

Reply:
No, fixed